quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/integration/jaxrs/TransactionalResponseInterceptor.java

```diff
@@ -34,12 +36,18 @@
 @ConstrainedTo(RuntimeType.SERVER)
 @Priority(10000)
 public class TransactionalResponseInterceptor implements WriterInterceptor, TransactionalSessionHandler {
+
     @Override
-    public void aroundWriteTo(WriterInterceptorContext context) throws IOException, WebApplicationException {
+    public void aroundWriteTo(WriterInterceptorContext context) {
         KeycloakSession session = Resteasy.getContextData(KeycloakSession.class);
+        OutputStream outputStream = context.getOutputStream();
 
+        // see https://github.com/keycloak/keycloak/issues/20451
         try {
+            context.setOutputStream(new BufferedOutputStream(outputStream, 8192));
```

Comment:
This won't work if the amount of bytes written by the interceptor chain reaches the buffer limit. The buffer will be flushed and you will still face the same problem.

That is why I ended up not wrapping the original output stream